### PR TITLE
Gracefully require PHP 5.6.0

### PR DESCRIPTION
**Pull request by pyronaur**
Fills #384

I moved the whole Co-Authors-Plus plugin code from `co-authors-plus.php` to `co-authors-plus-core.php` to ensure that the plugin gracefully deactivates when the PHP Version requirements aren't met. This way PHP 5.6 code can be used throughout the plugin, except `co-authors-plus.php` which has to maintain PHP 5.2.4 compatibility.

If the current user visiting the site doesn't have permissions to deactivate plugins - no message will be shown, but the plugin won't load.

As soon as a user with `deactivate_plugins` privileges visits the dashboard the plugin will auto-deactivate and display a message that Co-Authors-Plus requires PHP 5.6.

I've also updated the readme `.md` and `.txt` files. This will also display the PHP version requirement on the WordPress.org repo due to the `Require PHP` tag [that's now available](https://make.wordpress.org/plugins/2017/08/29/minimum-php-version-requirement/)

**TL;DR**

☑️  Graceful degradation
☑️  Updated Readme for users
☑️  Updated Readme for WordPress.org "Requires PHP"
☑️  Travis Config is already using PHP 5.6+

### CODE REVIEW

File: class-cap-requirements.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		public function auto_deactivate() {
+
+			if ( function_exists( 'deactivate_plugins' ) && current_user_can( 'deactivate_plugins' ) ) {
+				deactivate_plugins( $this->plugin_name );
+			}
+
+		}
+
+	}
+}
```
Comment (GaryJones) [inline class-cap-requirements.php]: Clear linespace needed at the end of the file here.

File: co-authors-plus.php [hunk 2]
```diff
@@ -32,1752 +32,17 @@
...
- * @return array Array of coauthor WP_Term objects
- */
-function cap_get_coauthor_terms_for_post( $post_id ) {
-	global $coauthors_plus;
-	return $coauthors_plus->get_coauthor_terms_for_post( $post_id );
-}
+// Load the plugin if PHP Requirements are met
+if ( $cap_requirements->enviroment_meets_requirements() ) {
+	require_once dirname( __FILE__ ) . '/co-authors-plus-core.php';
+}
```
Comment (GaryJones) [inline co-authors-plus.php]: Clear line space needed here.

**Comment by nielslange** [comment]
Closing this issue as the corresponding rule set had been updated accordingly in #751.
